### PR TITLE
Add CaseConversion commands to the sublime keymap.

### PR DESCRIPTION
**Pull request by WillOursler**
Hi! First time trying this sort of thing, apologies if I've screwed something up.

I talked to Maksym Taran internally about how best to add this feature to codemirror. He suggested the sublime keymap was most appropriate. Please let me know if I should restructure it another way.

Thanks! ~Will

### CODE REVIEW

File: keymap/sublime.js [hunk 1]
```diff
@@ -480,6 +480,112 @@
...
+  // repeated--delimiters--are--ignored -> [repeated, delimiters, are, ignored]
+  function splitStringToRedoCase(str) {
+    return str
+      // This part splits up camelCased portions of the string,
+      // seperating them with whitespace.
+      .replace(/([a-z])([A-Z])/g, '$1 $2')
+      .replace(/([A-Z])([a-z])/g, ' $1$2')
+      // Trim out any leading or trailing whitespace
+      // The replaces above sometimes produce some.
+      .trim()
```
Comment (marijnh) [inline keymap/sublime.js]: We're still targeting ES3 runtimes, so I'm afraid you'll have to do the trimming manually.

File: keymap/sublime.js [hunk 2]
```diff
@@ -480,6 +480,112 @@
...
+
+  function toUpperCaseFxn(word) {
+    return word.toUpperCase();
+  }
+
+  // These commands are designed to imitate the popular
+  // Case Conversion sublime plugin.
+  cmds['snakeCaseAtCursor'] = function(cm) {
+    modifyWordOrSelection(cm, function(str) {
+      return splitStringToRedoCase(str).map(toLowerCaseFxn).join('_');
```
Comment (marijnh) [inline keymap/sublime.js]: Again, ES3, so you'll need to write a `map` helper function.

**Comment by marijnh** [comment]
I added a few small comments, and I have one more general reservation -- this doesn't seem to be functionality that Sublime Text actually provides, so I don't think it belongs in the Sublime Text bindings. I guess the `modifyWordOrSelection` is helpful to have access to, but I don't think that's enough justification to put this here.

Additionally, we have a rule of not accepting new functionality into the distribution that could be created as an addon. Is copy-pasting `modifyWordOrSelection`, along with your new code, into a separate addon and putting it on NPM as, say, `codemirror-case-conversion`, an option for you?

**Comment by mtaran-google** [comment]
I think it'll be easier to have this be internal to our project (since publishing an open source repo involves some bureaucracy. This way we can also use new ES features easily :)

Thanks for taking a look, though!

**Comment by marijnh** [comment]
Great, thanks for understanding.

**Pull request closed without merging**